src/editorwidget/widget.ts

```diff
@@ -78,6 +110,166 @@ class EditorWidget extends CodeMirrorWidget {
       }
     });
   }
+
+  createMenu(layout : PanelLayout, editor : any) {
```

Comment:
This should be a `protected` method that returns a menu (since it is only intended to be called from within the class but could be overridden in a subclass), and the editor is of `CodeMirror.Editor` type.  Also missing documentation.